src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.Exceptions.cs

```diff
@@ -78,7 +78,7 @@ public async Task LeadingReferenceMetadataNotSupported()
 
             string exStr = ex.ToString();
             Assert.Contains("System.Text.Json.Serialization.Tests.ConstructorTests+Employee", exStr);
-            Assert.Contains("$.$id", exStr);
+            Assert.Contains("$.Manager.$ref", exStr);
```

Comment:
shouldn't it throw on `$id` already? (so basically expected path as it was) if it's not supported than why do we need to continue until we try to resolve `$ref`?

Reply:
This change is related to addressing #66604. Effectively we're relaxing support for reference metadata in classes with constructors _except_ in cases where the constructor arguments contain reference metadata (because cyclic objects cannot be deserialized using constructors). Because of that relaxation, the error is surfaced when the constructor arguments are being deserialized.